pkg/util/http/client.go

```diff
@@ -31,86 +27,68 @@ func SanitizeURL(message string) string {
 	return apiKeyRegExp.ReplaceAllString(message, apiKeyReplacement)
 }
 
-// CreateHTTPTransport creates an *http.Transport for use in the agent
-func CreateHTTPTransport() *http.Transport {
-	tlsConfig := &tls.Config{
-		InsecureSkipVerify: config.Datadog.GetBool("skip_ssl_validation"),
+// Client wraps (http.Client).Do and resets the underlying connections at the
+// configured interval
+type Client struct {
+	httpClient    *http.Client
+	newHTTPClient func() *http.Client
+	resetInterval time.Duration
+
+	lastReset time.Time
+	mutex     sync.RWMutex
+}
+
+// NewClient returns an initialized Client resetting connections at the passed resetInterval.
+// The underlying http.Client used will be created using the passed http client factory.
+func NewClient(resetInterval time.Duration, newHTTPClient func() *http.Client) *Client {
+	return &Client{
+		resetInterval: resetInterval,
+		httpClient:    newHTTPClient(),
+		newHTTPClient: newHTTPClient,
```

Comment:
I think the initial value of `lastReset` should be set here, not in `shouldReset`.

Reply:
addressed